File: PF with Python Jan 2022/15. Functions - Exercise/04. odd_and_even_sum.py
# 4.Odd and Even Sum
# You will receive a single number. You should write a function that returns the sum of all even and all odd digits in a given number. The result should be returned as a single string in the format:
# "Odd sum = {sum_of_odd_digits}, Even sum = {sum_of_even_digits}"
# Print the result of the function on the console.
#
# Input
# 1000435
#
# Output
# Odd sum = 9, Even sum = 4

def sum_even_odd_digits(n):

    even_sum = 0
    odd_sum = 0

    for i in n:
        if i % 2 == 0:
            even_sum += i
        else:
            odd_sum += i

    print(f"Odd sum = {odd_sum}, Even sum = {even_sum}")

num = list(map(int, list(input())))

sum_even_odd_digits(num)

File: JS Front-End/06. Exercise Functions and Statements/04. odd-and-even-sum.js
function oddEvenSums(num) {
  let numDigits = num
    .toString()
    .split("")
    .map((numString) => Number(numString));
  let evenSum = 0;
  let oddSum = 0;
  for (const digit of numDigits) {
    if (digit % 2 === 0) {
      evenSum += digit;
    } else {
      oddSum += digit;
    }
  }
  console.log(`Odd sum = ${oddSum}, Even sum = ${evenSum}`);
}

oddEvenSums(1000435);
oddEvenSums(3495892137259234);
